clippy_lints/src/from_instead_of_into.rs

```diff
@@ -0,0 +1,140 @@
+                let after_bound = &bounds[position + 1];
+                after = Some(snippet(cx, after_bound.span().with_hi(last_bound.span().hi()), ".."));
+            }
+
+            match (before, after) {
+                (None, None) => unreachable!(),
+                (Some(b), None) => b,
+                (None, Some(a)) => a,
+                (Some(b), Some(a)) => b + " + " + a,
+            }
+        }
+
+        if let WherePredicate::BoundPredicate(wbp) = wp {
+            if_chain! {
+                let bounds = wbp.bounds;
+                if let Some(position) = bounds.iter().position(|b| is_target_generic_bound(cx, b));
+                let target_bound = &bounds[position];
+                if let Some(tr_ref) = target_bound.trait_ref();
+                if let Some(def_id) = tr_ref.trait_def_id();
+                if let Some(last_seg) = tr_ref.path.segments.last();
+                if let Some(generic_arg) = last_seg.args().args.first();
+                if let Some(bounded_ty) = snippet_opt(cx, wbp.bounded_ty.span);
+                if let Some(generic_arg_of_from_or_try_from) = snippet_opt(cx, generic_arg.span());
+                then {
+                    let replace_trait_name;
+                    let target_trait_name;
+                    if cx.tcx.is_diagnostic_item(sym::from_trait, def_id) {
+                        replace_trait_name = "Into";
+                        target_trait_name = "From";
+                    } else if cx.tcx.is_diagnostic_item(sym::try_from_trait, def_id) {
+                        replace_trait_name = "TryInto";
+                        target_trait_name = "TryFrom";
+                    } else {
+                        replace_trait_name = "";
+                        target_trait_name = "";
+                    }
+
+                    if !replace_trait_name.is_empty() && !target_trait_name.is_empty() {
+                        let message = format!("{} trait is preferable than {} as a generic bound", replace_trait_name, target_trait_name);
+
+                        let extracted_where_predicate = format!("{}: {}<{}>", generic_arg_of_from_or_try_from, replace_trait_name, bounded_ty);
+                        let sugg;
+                        if bounds.len() == 1 {
+                            sugg = extracted_where_predicate;
+                        } else {
+                            let bounds = get_reduced_bounds_str(cx, position, bounds);
+                            sugg = format!("{}, {}: {}", extracted_where_predicate, bounded_ty, bounds);
+                        }
+                        span_lint_and_sugg(
+                            cx,
+                            FROM_INSTEAD_OF_INTO,
+                            wp.span(),
+                            &message,
+                            "try",
+                            sugg,
+                            Applicability::MaybeIncorrect
+                        );
```

Comment:
So instead of trying to rebuild the whole predicate string without that one bound, it would be easier so just get the span of the bound and suggest to remove it. 

> But there are some part I still have doubt, when you refer to 2 part suggestion you are not referring to multi span suggestion, do I understand correctly?

No. You should use `span_lint_and_then` here instead and add 2 suggestions if there is more than one bound. Just grep for `span_lint_and_then` in the repo and you'll find how to do this.